rosbag_snapshot/src/snapshotter.cpp

```diff
@@ -123,8 +123,21 @@ void MessageQueue::clear()
 
 void MessageQueue::_clear()
 {
-  queue_.clear();
-  size_ = 0;
+  if (!queue_.empty()) {
+    SnapshotMessage latest = _pop_back();
+
+    queue_.clear();
+    size_ = 0;
+
+    ros::M_string::const_iterator it = latest.connection_header->find("latching");
+    if ((it != latest.connection_header->end()) && (it->second == "1")) {
+      // Restore the latest message for latched topics
+      _push(latest);
```

Comment:
I think we talked about this, but I now wonder if it makes sense to update the stamp on it. I think we do that in the bag recording for latched topics? I could be wrong. I just want to avoid the case where a robot has been running for days, and playing back the bump bag file shows a bag that has a length of hundreds of thousands of seconds.

Reply:
https://github.com/locusrobotics/ros_comm/blob/f7e9308ea2aafed50a09ddd07ba84603c933b900/tools/rosbag/src/recorder.cpp#L475-L477

Reply:
I could do `latest.time = ros::Time::now();` which would still result in a gap since this occurs when a bag is triggered so there will be a gap between here and the next trigger, but maybe that's acceptable? If the time should be updated when the message is bagged then that should be possible too. Just let me know which you prefer :)

Reply:
@ayrton04 My last two commits should fix the timestamp issue by doing two things:

1. Always include latched messages in the message queues that are <= the stop time of the trigger service request. Basically just ignore the start time in the service request for latched topics. See `snapshotter.cpp:254`
2. If any messages have timestamps older than the start time of the bag, set them to the start time. See `snapshotter.cpp:406`